### ISSUE TITLE
Default option in prop decorator ignored for array of refs

### ISSUE THREAD

**Issue by maxwellpeterson** [typegoose/typegoose#576]
## Versions

- System: Linux (Ubuntu 20.04)
- NodeJS: 14.16.1
- Typescript: 4.2.4
- Compiler / Transpiler: ts-node
- Typegoose(NPM): 7.6.3
- mongoose: 5.10.18
- mongodb: 4.4.6

## What is the Problem?

The `default` option in the `prop` decorator seems to be ignored when the field is an array of references. If `default` is set to `undefined` then an empty array is saved for that field if no value is provided at creation, when that field should instead be excluded from the saved document.

## Code Example

```ts
// NodeJS: 14.16.1
// MongoDB: 4.4.6
import { prop, Ref } from "@typegoose/typegoose"; // @typegoose/typegoose@7.6.1
import { User } from "other/local/file";

class Example {
  // If value for strings field is not provided at creation,
  // then strings field does not appear in saved document
  // (this is expected behavior)
  @prop({ type: [String], default: undefined })
  public strings?: string[];

  // If value for users field is not provided at creation,
  // then users field is an empty array in saved document
  // (this is unexpected behavior)
  @prop({ ref: User, default: undefined })
  public users?: Ref<User>[];
}
```

## Do you know *why* it happens?

*no*

---

**Comment by hasezoey** [comment]
did you already try instead of using `default` directly, to put it in [`outerOptions: { default }`](https://typegoose.github.io/typegoose/docs/api/decorators/prop#outeroptions)?

when this works, then its because both types `Array` and `Reference` have the ability to hold the option `default`, and typegoose only has one at the top level, and chooses the lower type from the array (in this case `Reference`)

Note: i know that the names do not correspond to either `mongoose.Schema.Types` or `mongoose.Types`, its just to simplify

**Comment by maxwellpeterson** [comment]
I hadn't tried `outerOptions` but adding this didn't change the behavior. This was the updated decorator that I tried:
```ts
@prop({ ref: User, outerOptions: { default: undefined }})
public users?: Ref<User>[];
```
An empty array still gets added to the field if no value is provided at creation.

**Comment by hasezoey** [comment]
sorry, i didnt read your question correctly

i could reproduce it in typegoose `7.6.3` and mongoose `5.10.18`, but it seems fixed in typegoose `8.0.0-beta.x` and mongoose `5.13.2`

**Comment by hasezoey** [comment]
i looked again, it is fixed since b39eff9367e6910515944a140d7c939dddc532e3 in beta, it will not be in 7.x because it is an breaking change

closing as fixed in beta and wontfix in 7.x
